### PR TITLE
Refactor on NodeListTree

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

This is a smaller refactor. Previously, the logic for determining which nodes were disabled due to modular pipelines was duplicated in both the NodeListTree component  and the `getNodeDisabled` selector. To improve maintainability and reduce redundancy, the `getnodesDisabledViaModularPipeline` logic was extracted and made into it's own selector. Now, this logic is shared and reused by both the NodeListTree component and the `getNodeDisabled` selector.

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/components/nodes-panel/nodes-panel.test.js [hunk 1]
```diff
@@ -251,10 +251,6 @@ describe('NodesPanel', () => {
         .find('.node-list-tree-item-row')
         .map((row) => [row.prop('title'), !row.hasClass('row--disabled')]);
 
-    const elementsEnabled = (wrapper) => {
```
Comment (rashidakanchwala) [inline src/components/nodes-panel/nodes-panel.test.js:254]: This was not being used anywhere and throwing eslint error hence removed it.

File: src/selectors/disabled.js [hunk 2]
```diff
@@ -63,6 +63,57 @@ export const getNodeDisabledTag = createSelector(
     })
 );
 
+/**
+ * Determine if a node is disabled via disabled modular pipeline or focused modular pipeline.
+ */
+export const getnodesDisabledViaModularPipeline = createSelector(
```
Comment (jitu5) [inline src/selectors/disabled.js:69]: [Nit]
```suggestion
export const getNodesDisabledViaModularPipeline = createSelector(
```

**Review by jitu5 — APPROVED** [review]
LGTM!

**Review by Huongg — APPROVED** [review]
I tested the changes and they look pretty good to me. very nice thats the logic for disabled with Modular Pipeline has it own selector now

**Pull request merged**